Update CircleCI config to exclude target directory

meddle0x53 opened the pull request:
## Motivation

The semver checks fail too often, so maybe caching target is the reason for that. Removing this to try to fix the problem.

## Test Plan

Check if the CI passes for the semver checks.

## Documentation

Removed target directory from cache save paths.
This may fix the semver check fails.

meddle0x53 committed:
Update CircleCI config to exclude target directory

Removed target directory from cache save paths.
This may fix the semver check fails. 

Signed-off-by: meddle <n.tzvetinov@gmail.com>

vicsn commented:
Seems like the semver test still fails with "error: An object with id ffb3b2c89734c2c29ee87fd4db880fe3b46789dc could not be found".

I don't see the error message reported anywhere on github: https://github.com/obi1kenobi/cargo-semver-checks/issues

Can you try renaming the cache key so it clears:           cache_key: v4.2.0-rust-1.88.0-cargo-semver-checks-cache

kaimast commented:
Maybe this helps: the object is `.circleci/config.yml`. It is referenced in recent commits, including some on `staging`. My initial guess that it was due to some deleted branch, but that does not seem to be the case.

meddle0x53 committed:
Update cache key for semver checks in CircleCI config

Signed-off-by: meddle <n.tzvetinov@gmail.com>

meddle0x53 committed:
Update cache keys in CircleCI config

Signed-off-by: meddle <n.tzvetinov@gmail.com>

meddle0x53 committed:
Update CircleCI config to manage git cache

Comment out the git cache path and add a step to remove cached git checkouts.

Signed-off-by: meddle <n.tzvetinov@gmail.com>

meddle0x53 committed:
Enhance semver-checks script with error handling

Signed-off-by: meddle <n.tzvetinov@gmail.com>

meddle0x53 committed:
Change git checkout method to full

Signed-off-by: meddle <n.tzvetinov@gmail.com>

meddle0x53 committed:
Change checkout method to full in CircleCI config

Signed-off-by: meddle <n.tzvetinov@gmail.com>

meddle0x53 committed:
Refactor CircleCI config for cargo-machete installation

Updated CircleCI config to conditionally install cargo-machete and adjusted cache keys.

Signed-off-by: meddle <n.tzvetinov@gmail.com>

meddle0x53 committed:
Merge branch 'staging' into meddle0x53-patch-1

meddle0x53 committed:
Refactor semver-checks.sh to simplify git commands

Removed unnecessary git fetch commands from the script.

Signed-off-by: meddle <n.tzvetinov@gmail.com>

meddle0x53 commented on .circleci/config.yml:
```diff
@@ -1088,7 +1091,12 @@ jobs:
+      - run:
+          name: Drop cached git checkouts (semver)
+          command: |
+            rm -rf /home/circleci/.cargo/git
```
The combination of the full checkout and this fixes the missing revision.

meddle0x53 replied:
I can try returning the cache of the git folder here to see if it will work with the full checkout only.

vicsn replied:
sounds good, yes let's make the diff as small as possible

kaimast commented on .circleci/config.yml:
```diff
@@ -163,7 +163,7 @@ commands:
             - /home/circleci/.cache/sccache
             - /home/circleci/.cargo/registry
-            - /home/circleci/.cargo/git
+            #- /home/circleci/.cargo/git
```
Should we add a comment here why this is not cached?

meddle0x53 replied:
I think just removing it once fixes the revision problem, so will return it to see how it goes.

meddle0x53 committed:
Uncomment git cache path in CircleCI config

Signed-off-by: meddle <n.tzvetinov@gmail.com>

meddle0x53 committed:
Merge branch 'staging' into meddle0x53-patch-1

meddle0x53 commented:
OK, so there were 2 problems:
1. The script was not checking out everything so we were getting the revision error. I fixed it by making sure the repo is fully cloned for this check.
2. Cargo machete was already there, because of the cache so I did a change to only install it if it is not there.

meddle0x53 committed:
Remove cached git checkouts step from config

Removed step to drop cached git checkouts in CircleCI.

Signed-off-by: meddle <n.tzvetinov@gmail.com>

kaimast reviewed (approved):
LGTM!

Maybe we can squash the commits? 13 commits for an 6 line change is quite a lot.

meddle0x53 committed:
Update CircleCI config to run the semver workflow

kaimast committed:
Merge pull request #3033 from meddle0x53/meddle0x53-patch-1

Update CircleCI config to run the semver workflow

kaimast commented:
I cherry-picked the most recent commit onto #2999 and will merge that one, once its tests have passed.
